### PR TITLE
fix da.pad example for numpy 1.20

### LINKED ISSUE #4858: doctest failure with numpy 1.20
**What happened**:

Our doctests fail since numpy 1.20 came out:

https://github.com/pydata/xarray/pull/4760/checks?check_run_id=1818512841#step:8:69

**What you expected to happen**:

They don't ;-)

**Minimal Complete Verifiable Example**:

The following fails with numpy 1.20 while it converted `np.NaN` to an integer before ([xarray.DataArray.pad](http://xarray.pydata.org/en/v0.16.2/generated/xarray.DataArray.pad.html#xarray.DataArray.pad) at the bottom)

```python
import numpy as np

x = np.arange(10)
x = np.pad(x, 1, "constant", constant_values=np.nan)
```
requires numpy 1.20

**Anything else we need to know?**:

- that's probably related to https://numpy.org/doc/stable/release/1.20.0-notes.html#numpy-scalars-are-cast-when-assigned-to-arrays
- I asked if this behavior will stay: https://github.com/numpy/numpy/issues/16499#issuecomment-772342087
- One possibility is to add a check `np.can_cast(constant_values.dtype, array.dtype)` (or similar) for a better error message.

### PR DESCRIPTION
**Pull request by mathause**
- [x] Closes #4858
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Review by dcherian — APPROVED** [review]

**Comment by mathause** [comment]
cc @mark-boer

File: xarray/core/dataarray.py [hunk 1]
```diff
@@ -3880,16 +3879,16 @@ def pad(
           * x        (x) float64 nan 0.0 1.0 nan
           * y        (y) int64 10 20 30 40
             z        (x) float64 nan 100.0 200.0 nan
-        >>> da.pad(x=1, constant_values=np.nan)
+
+        Careful, ``constant_values`` can be coerced to the data type of the array which
+        may lead to a loss of precision:
```
Comment (keewis) [inline xarray/core/dataarray.py:3884]: is this unconditional?
```suggestion
        Careful, ``constant_values`` are coerced to the data type of the array which
        may lead to a loss of precision:
```

**Review by keewis — COMMENTED** [review]
could you add a test for the correct behavior? It's somewhat unfortunate that this was only caught by the doctests.

**Comment by mark-boer** [comment]
@keewis, I agree a test would be nice, but what kind of test would you want to add? I mean, xarray just mimics the behaviour of `np.pad` and `dask.array.pad`. And the default behaviour of numpy was a bit funky, which is probably why they changed it.

**Comment by keewis** [comment]
not sure. I would probably try to either replicate the new behavior of `numpy` with an example, or just copy the doctest into a test.

**Comment by mathause** [comment]
Thanks for the review. I added some minimal tests. I think it could be good to discuss the type coercion in a new issue.

**Review by keewis — APPROVED** [review]
thanks, looks good to me now. :+1: for a new issue.

**Pull request merged**